src/trio/_core/_parking_lot.py

```diff
@@ -234,6 +272,34 @@ def repark_all(self, new_lot: ParkingLot) -> None:
         """
         return self.repark(new_lot, count=len(self))
 
+    def break_lot(self, task: Task | None = None) -> None:
+        """Break this lot, causing all parked tasks to raise an error, and any
+        future tasks attempting to park to error. Unpark & repark become no-ops as the
+        parking lot is empty.
+        The error raised contains a reference to the task sent as a parameter.
+        """
```

Comment:
Summary line here too. Also "The error raised contains a reference to the task sent as a parameter." should probably just mention that `broken_by` is available on parking lots, because I personally missed that previously.

Reply:
not super happy with the docstring, but should be fine now? Feel free to suggest changes